src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -46,39 +73,275 @@ static ColliderCylinderInit D_80BA1B94 = {
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -1200, ICHAIN_CONTINUE), ICHAIN_F32_DIV1000(minVelocityY, -20000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 150, ICHAIN_CONTINUE),   ICHAIN_F32(uncullZoneForward, 900, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 100, ICHAIN_CONTINUE),   ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
+};
+
+void ObjTsubo_SpawnCollectible(ObjTsubo* this, GlobalContext* globalCtx) {
+    s16 dropParams;
+
+    dropParams = this->actor.params & 0x1F;
+    if ((dropParams >= 0) && (dropParams < 0x1A)) {
+        Item_DropCollectible(globalCtx, &this->actor.world.pos,
+                             (dropParams | (((this->actor.params >> 9) & 0x3F) << 8)));
+    }
+}
+
+void ObjTsubo_Fall(ObjTsubo* this) {
+    this->actor.velocity.y += this->actor.gravity;
+    if (this->actor.velocity.y < this->actor.minVelocityY) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+s32 ObjTsubo_SnapToFloor(ObjTsubo* this, GlobalContext* globalCtx) {
+    CollisionPoly* floorPoly;
+    Vec3f pos;
+    s32 bgID;
+    f32 floorY;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y + 20.0f;
+    pos.z = this->actor.world.pos.z;
+    floorY = BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &floorPoly, &bgID, &this->actor, &pos);
+    if (floorY > BGCHECK_Y_MIN) {
+        this->actor.world.pos.y = floorY;
+        Math_Vec3f_Copy(&this->actor.home.pos, &this->actor.world.pos);
+        return true;
+    } else {
+        osSyncPrintf("地面に付着失敗\n");
+        return false;
+    }
+}
+
+void ObjTsubo_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTsubo* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80BA1B94);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+}
+
+void ObjTsubo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTsubo* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ObjTsubo_InitCollider(&this->actor, globalCtx);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    if (ObjTsubo_SnapToFloor(this, globalCtx) == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->objTsuboBankIndex = Object_GetIndex(&globalCtx->objectCtx, *(sObjectIds + (((this->actor.params >> 8) & 1))));
+    if (this->objTsuboBankIndex < 0) {
+        osSyncPrintf("Error : バンク危険！ (arg_data 0x%04x)(%s %d)\n", this->actor.params, "../z_obj_tsubo.c", 0x19A);
+        Actor_Kill(&this->actor);
+    } else {
+        func_80BA152C(this);
+        osSyncPrintf("(dungeon keep 壷)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void ObjTsubo_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjTsubo* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
```

Comment:
small thing but we like to put line breaks after the declarations. may be relevant in a few other places in the file